### ISSUE TITLE
Receiving disconnect event in Browser and Backend

### ISSUE THREAD

**Issue by ZuSe** [twilio/twilio-video.js#147]
- [X] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [X ] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [X] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.

**Expected behavior:**
There should be no disconnect event triggered unless a participant actively leaves a room

**Actual behavior:**
Hey folks, I receive on my backend the following webhook from twilio:

Communication <QueryDict: {'RoomName': ['c5b44c14-2ca3-4224-b0e7-54388f96b852'], 'StatusCallbackEvent': ['participant-disconnected'], 'RoomStatus': ['in-progress'], 'ParticipantIdentity': ['16'], 'Timestamp': ['2017-08-09T09:18:04.195Z'], 'ParticipantSid': ['PA1fddfabdfeded034e9997c60869b3d05'], 'SequenceNumber': ['7'], 'RoomSid': ['RM79615c6d57206a6c1f80b8018895e220'], 'ParticipantStatus': ['disconnected'], 'ParticipantDuration': ['0'], 'AccountSid': ['xxxx]}>

this is caused by the websocket being disconnected in the web client:

```
Wed Aug 09 2017 12:00:10 GMT+0200 (CEST) | sip.transaction.nist | Timer J expired for non-INVITE server transaction z9hG4bK8e4a65f2-4cc8-420d-87d5-d19da6cc0e5d_6772d868_275-331396360501912715
twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:10 GMT+0200 (CEST) | sip.transport | received WebSocket text message:

SIP/2.0 481 Call leg/Transaction does not exist
CSeq: 6491 BYE
Call-ID: 0p9m3ecq4bkbpfg82iv3
From: <sip:eb6e45be-7113-4723-b5ba-a6abc1323aa7@endpoint.twilio.com>;tag=c2jmmpups2
To: <sip:orchestrator@endpoint.twilio.com>;tag=83460691_6772d868_8e4a65f2-4cc8-420d-87d5-d19da6cc0e5d
Via: SIP/2.0/WSS 127.0.0.1:37908;branch=z9hG4bK5499680;rport=37908
Server: Twilio
Content-Length: 0



twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:10 GMT+0200 (CEST) | sip.transport | closing WebSocket wss://endpoint.twilio.com
twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:10 GMT+0200 (CEST) | sip.transport | received WebSocket text message:

INFO sip:35pk1bo8@127.0.0.1:57066;transport=wss;ob SIP/2.0
CSeq: 1 INFO
From: <sip:orchestrator@endpoint.twilio.com>;tag=17120598_6772d868_427a2fca-2db5-4764-a3f2-e54454aa37aa
To: <sip:6e88043e-948c-4a4f-a1aa-e2642369deab@endpoint.twilio.com>;tag=2ghp9ug9qo
Call-ID: denon6p27lnjehjkgt37
Max-Forwards: 69
Via: SIP/2.0/WSS 172.18.6.121:443;branch=z9hG4bK427a2fca-2db5-4764-a3f2-e54454aa37aa_6772d868_176-8152825070525891452
Content-Type: application/room-signaling+json
Recv-Info: room-signaling
Info-Package: room-signaling
Supported: room-signaling
User-Agent: Twilio Gateway
X-Twilio-CallSid: CAa4bcd09a7bbc5ee89213f2e0ad2d8a82
Content-Length: 879

{"version":1,"type":"update","sid":"RM35ccd8007881caeb6168872bf39cc234","name":"RM35ccd8007881caeb6168872bf39cc234","participant":{"sid":"PAcc3421268e4e0bd882feb12d24d1b786","identity":"16","tracks":[{"kind":"audio","id":"a9e2eb8d-bc5a-4b6b-b5ba-d58413b0acde","enabled":true,"sid":"MT72884ac7486de219b445914551773f5d"},{"kind":"video","id":"d7f70cc6-7164-4d94-ae9c-897d7948889d","enabled":true,"sid":"MTc085224a8a78e386d77cd1695b05b78d"}],"revision":1,"state":"connected"},"participants":[{"sid":"PAa901c353e1fddd08c9f2b0c1ea93053e","identity":"16","tracks":[{"kind":"audio","id":"a9e2eb8d-bc5a-4b6b-b5ba-d58413b0acde","enabled":true,"sid":"MT2d69ace38e54a48baf16496e5748a61f"},{"kind":"video","id":"d7f70cc6-7164-4d94-ae9c-897d7948889d","enabled":true,"sid":"MT81f6de0c7b13e772f7ed6983e145acf5"}],"revision":2,"state":"disconnected"}],"recording":{"enabled":false,"revision":1}}

twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:10 GMT+0200 (CEST) | sip.transport | sending WebSocket message:

SIP/2.0 200 OK
Via: SIP/2.0/WSS 172.18.6.121:443;branch=z9hG4bK427a2fca-2db5-4764-a3f2-e54454aa37aa_6772d868_176-8152825070525891452
To: <sip:6e88043e-948c-4a4f-a1aa-e2642369deab@endpoint.twilio.com>;tag=2ghp9ug9qo
From: <sip:orchestrator@endpoint.twilio.com>;tag=17120598_6772d868_427a2fca-2db5-4764-a3f2-e54454aa37aa
Call-ID: denon6p27lnjehjkgt37
CSeq: 1 INFO
Supported: outbound, room-signaling, timer
User-Agent: SIP.js/0.7.5
Content-Length: 0



twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:10 GMT+0200 (CEST) | sip.transport | received WebSocket text message:

INFO sip:35pk1bo8@127.0.0.1:57066;transport=wss;ob SIP/2.0
CSeq: 2 INFO
From: <sip:orchestrator@endpoint.twilio.com>;tag=17120598_6772d868_427a2fca-2db5-4764-a3f2-e54454aa37aa
To: <sip:6e88043e-948c-4a4f-a1aa-e2642369deab@endpoint.twilio.com>;tag=2ghp9ug9qo
Call-ID: denon6p27lnjehjkgt37
Max-Forwards: 69
Via: SIP/2.0/WSS 172.18.6.121:443;branch=z9hG4bK427a2fca-2db5-4764-a3f2-e54454aa37aa_6772d868_190-6120811616058564153
Content-Type: application/room-signaling+json
Recv-Info: room-signaling
Info-Package: room-signaling
Supported: room-signaling
User-Agent: Twilio Gateway
X-Twilio-CallSid: CAa4bcd09a7bbc5ee89213f2e0ad2d8a82
Content-Length: 645

{"version":1,"type":"update","peer_connections":[{"id":"833a8d13-19ea-45ef-90df-a62b623b49db","description":{"type":"close","revision":2}}],"sid":"RM35ccd8007881caeb6168872bf39cc234","name":"RM35ccd8007881caeb6168872bf39cc234","participant":{"sid":"PAcc3421268e4e0bd882feb12d24d1b786","identity":"16","tracks":[{"kind":"audio","id":"a9e2eb8d-bc5a-4b6b-b5ba-d58413b0acde","enabled":true,"sid":"MT72884ac7486de219b445914551773f5d"},{"kind":"video","id":"d7f70cc6-7164-4d94-ae9c-897d7948889d","enabled":true,"sid":"MTc085224a8a78e386d77cd1695b05b78d"}],"revision":1,"state":"connected"},"participants":[],"recording":{"enabled":false,"revision":1}}

twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:10 GMT+0200 (CEST) | sip.transport | sending WebSocket message:

SIP/2.0 200 OK
Via: SIP/2.0/WSS 172.18.6.121:443;branch=z9hG4bK427a2fca-2db5-4764-a3f2-e54454aa37aa_6772d868_190-6120811616058564153
To: <sip:6e88043e-948c-4a4f-a1aa-e2642369deab@endpoint.twilio.com>;tag=2ghp9ug9qo
From: <sip:orchestrator@endpoint.twilio.com>;tag=17120598_6772d868_427a2fca-2db5-4764-a3f2-e54454aa37aa
Call-ID: denon6p27lnjehjkgt37
CSeq: 2 INFO
Supported: outbound, room-signaling, timer
User-Agent: SIP.js/0.7.5
Content-Length: 0



twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:10 GMT+0200 (CEST) | sip.transaction.nist | Timer J expired for non-INVITE server transaction z9hG4bK427a2fca-2db5-4764-a3f2-e54454aa37aa_6772d868_176-8152825070525891452
twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:10 GMT+0200 (CEST) | sip.transaction.nist | Timer J expired for non-INVITE server transaction z9hG4bK427a2fca-2db5-4764-a3f2-e54454aa37aa_6772d868_190-6120811616058564153
twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:12 GMT+0200 (CEST) | sip.transport | WebSocket disconnected (code: 1006)
twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:12 GMT+0200 (CEST) | sip.transport | WebSocket abrupt disconnection
LoggerFactory.print @ twilio-video.js?1502264633629:13084
LoggerFactory.(anonymous function) @ twilio-video.js?1502264633629:13101
Logger.(anonymous function) @ twilio-video.js?1502264633629:13095
onClose @ twilio-video.js?1502264633629:18754
ws.onclose @ twilio-video.js?1502264633629:18697
wrapFn @ zone.js?1502264633624:1056
ZoneDelegate.invokeTask @ zone.js?1502264633624:424
onInvokeTask @ ng_zone.ts:257
ZoneDelegate.invokeTask @ zone.js?1502264633624:423
Zone.runTask @ zone.js?1502264633624:191
ZoneTask.invoke @ zone.js?1502264633624:486
twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:12 GMT+0200 (CEST) | sip.ua | connection state set to 1
twilio-video.js?1502264633629:13084 Wed Aug 09 2017 12:00:12 GMT+0200 (CEST) | sip.transaction.ict | transport error occurred, deleting INVITE client transaction z9hG4bK5072411
```

I can't explain why it does it. When a second client joins the room the p2p connection is established.
However, since we receive the disconnect event our overview dashboard is somehow broken because our application assumes that room is empty.

**Software versions:**

 - [x] Browser(s): Chrome 59
 - [x] Operating System: Ubuntu 16.04.02 (also tested with macOS Sierra)
 - [x] twilio-video.js: 1.2.0
 - [x] Third-party libraries (e.g., Angular, React, etc.): Angular4 4.3.3

**Comment by ZuSe** [comment]
For some reason we joined the room twice. all fine now.

**Comment by mjamal-tiu** [comment]
The same error I am now getting at the Chrome browser and it is working fine in firefox.

A week ago it was working fine. Please let me know if someone has resolved this issue.

**Comment by manjeshbhargav** [comment]
Hi @mjamal-tiu ,

Thanks for writing in with your issue. Can you give us a Room SID which we can look at? 

Thanks,
Manjesh Malavalli
JSDK Team

**Comment by mjamal-tiu** [comment]
Issue is resolved based on suggestion provided on #576
